dqc_us_rules/dqc_us_0033_0036.py

```diff
@@ -117,17 +121,24 @@ def _doc_period_end_date_check(model_xbrl):
             # If the DocumentPeriodEndDate context check doesn't fire,
             # we will check all dei fact context end dates against it.
             for fact in fact_group:
-                if ((fact.contextID in not_valid_dped or
-                     fact.context is None or
+                if ((fact.context is None or
                      fact.context.endDatetime is None or
                      fact.concept.periodType != 'duration'
                      )):
                     continue
 
-                if context_eop_date != dateunionDate(
-                    fact.context.endDatetime,
-                    subtractOneDay=True
-                ):
+                should_break = False
+                for fact_axis, fact_dim_value in fact.context.segDimValues.items():
+                    if fact_dim_value.memberQname.localName in not_valid_dped:
+                        should_break = True
```

Comment:
Since this breaks to a continue, you should ensure that this variable gets initialized to False at the top of the loop. Also, since this does signal whether or not to continue, the variable should probably be named "should_continue" or "should_skip".